src/ImageSharp/Memory/Allocators/MemoryAllocator.cs

```diff
@@ -22,15 +34,15 @@ public abstract class MemoryAllocator
 
     /// <summary>
     /// Gets or sets the maximum allowable allocatable size of a 2 dimensional buffer.
-    /// Defaults to <value>65535 * 65535.</value>
+    /// Defaults to <value><see cref="DefaultMaxAllocatableSize2DInBytes"/>.</value>
     /// </summary>
-    public Size MaxAllocatableSize2D { get; set; } = new Size(ushort.MaxValue, ushort.MaxValue);
+    public Size MaxAllocatableSize2DInBytes { get; set; } = DefaultMaxAllocatableSize2DInBytes;
```

Comment:
We need it to somehow reflect the actual dimensions of a buffer (to the user an image). A memory group is an implementation detail but if I can say don't allow a total allocation of more than X, Y then the groups are covered.

Reply:
I would prefer to see it the other way around. If you define a limit for a discontigous allocation it will cover the image limit.

> A memory group is an implementation

Not entirely, we have documented that our buffers are discontiguous and `IMemoryGroup<T>` is public. Regardless, we don't need to involve those concepts to when defining the limits. We can just document that maximum in-memory image size is `X (mega/giga/tera)bytes`.

Reply:
If we believe we need dimension limits (eg. bc various formats have them anyways), we can define them separately either in `Image<T>` or in `Buffer2D<T>` code, without involving `sizeof(TPixel)`. But for me that concern is orthogonal to the issue this PR is aimed to fix.

Reply:
The issue directly stems from attempt to allocate an image that is too large. Fixing any additional allocation limits was a something we needed to do in addition to that public facing allocation. 

By placing a limit at the point of allocation call, before we start even looking at memory groups, I avoid the possibility of over allocation in all cases. The limits are not dimensional limits, they reflect dimensional limits. We can now say, we've defaulted to an image of X*Y at this pixel format bit depth. It's a starting point that users can understand and update when required.

Whether our buffers are discontiguous doesn't really change things.  We've already established that attempting to allocate an image of overly large dimensions will cause issues.